atom/browser/web_contents_preferences.cc

```diff
@@ -293,12 +294,15 @@ void WebContentsPreferences::AppendCommandLineSwitches(
   if (IsEnabled(options::kWebviewTag))
     command_line->AppendSwitch(switches::kWebviewTag);
 
+  bool sandbox =
+      is_subframe && !IsEnabled(options::kNodeIntegrationInSubFrames);
+
   // If the `sandbox` option was passed to the BrowserWindow's webPreferences,
   // pass `--enable-sandbox` to the renderer so it won't have any node.js
   // integration.
-  if (IsEnabled(options::kSandbox)) {
+  if (IsEnabled(options::kSandbox) || sandbox) {
     command_line->AppendSwitch(switches::kEnableSandbox);
-  } else if (!command_line->HasSwitch(switches::kEnableSandbox)) {
+  } else if (!command_line->HasSwitch(switches::kEnableSandbox) && !sandbox) {
```

Comment:
Could you perhaps add some comments around this logic, it took me a few reads to wrap my head around what we were doing here.  Would be great to understand at a glance

Reply:
Actually, I think this condition should not be changed.